core/arch/arm/kernel/mutex.c

```diff
@@ -81,6 +110,27 @@ static void __mutex_unlock(struct mutex *m, const char *fname, int lineno)
 	wq_wake_next(&m->wq, m, fname, lineno);
 }
 
+static void __mutex_unlock_recursive(struct recursive_mutex *m,
+				     const char *fname, int lineno)
+{
+	assert_have_no_spinlock();
+	assert(thread_get_id_may_fail() != -1);
```

Comment:
s/-1/THREAD_ID_INVALID/ for consistency
Since there are other places where -1 is used and could be replaced with `THREAD_ID_INVALID`, maybe this can be done in a later P-R.